src/renderer/routes/test_sequencer_panel/utils/SequenceHandler.ts

```diff
@@ -20,135 +19,174 @@ export function Err<T, E>(error: E): Result<T, E> {
 // End of note ------------------------------------------
 
 import {
-  readJsonProject,
-  stringifyProject,
+  readJsonSequence,
+  stringifySequence,
 } from "@/renderer/routes/test_sequencer_panel/utils/TestSetUtils";
 import {
+  TestSequenceContainer,
   TestSequenceElement,
   TestSequencerProject,
 } from "@/renderer/types/test-sequencer";
 import { createNewTest } from "@/renderer/hooks/useTestSequencerState";
 
 // Exposed API
 export type StateManager = {
-  elem: TestSequenceElement[];
-  setElems: (elems: TestSequenceElement[]) => void;
+  elems: TestSequenceElement[];
+  addNewSequence: (
+    project: TestSequencerProject,
+    elements: TestSequenceElement[],
+  ) => void;
+  removeSequence: (name: string) => void;
   project: TestSequencerProject | null;
-  setProject: (project: TestSequencerProject | null) => void;
-  setUnsaved: (unsaved: boolean) => void;
+  sequences: TestSequenceContainer[];
 };
 
-export async function createProject(
-  project: TestSequencerProject,
+export async function createSequence(
+  sequence: TestSequencerProject,
   stateManager: StateManager,
   throwInsteadOfResult: boolean = false,
 ): Promise<Result<null, Error>> {
   // Create a new project from the element currently in the test sequencer
   // - Will valide that each element is in the base folder
   try {
-    const elems = stateManager.elem;
-    project = validatePath(project);
-    console.log(project);
-    project = updateProjectElements(
-      project,
-      await createProjectElementsFromTestSequencerElements(
+    const elems = stateManager.elems;
+    sequence = validatePath(sequence);
+    sequence = updateSequenceElement(
+      sequence,
+      await createExportableSequenceElementsFromTestSequencerElements(
         elems,
-        project.projectPath,
+        sequence.projectPath,
         true,
       ),
     );
-    await saveToDisk(project);
-    await syncProject(project, stateManager);
+    await saveToDisk(sequence);
+    await syncSequence(sequence, stateManager);
     return Ok(null);
   } catch (e: unknown) {
     if (throwInsteadOfResult) throw e;
     return buildResultFromCatch(e);
   }
 }
```

Comment:
I don't agree in this case, this file uses the exception, it's actively throws exception as a way to cut the exception. I know we're not active fans of the the use exception now, but in this case I'd say changing it would make the file less readable.

The exposed API doesn't throw, but the inner logic does as a way to make the hole logic easy to understand

Reply:
WIP on refactoring the file to remove exception for neverthrow [here](https://github.com/LatentDream/flojoy-studio/pull/5/files)

Reply:
Merged